gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```
@@ -0,0 +1,398 @@
+    client.updateDatabaseParameters(dbName, optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> props) {
+    DatabaseName dbName = databaseName(namespace);
+    if (dbName == null) {
+      throwInvalidDbNamespaceError(namespace);
+    }
+
+    HiveDatabaseOptions.Builder optionsBuilder =
+        loadDatabase(namespace).toBuilder().getHiveOptionsBuilder();
+    props.forEach(optionsBuilder::removeParameters);
+    client.updateDatabaseParameters(dbName, optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      client.getCatalog(catalogName);
+      return ImmutableMap.of();
+    } else if (namespace.levels().length == 1) {
+      return loadDatabase(namespace).getHiveOptions().getParametersMap();
+    } else {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  protected Map<String, String> properties() {
+    return properties == null ? ImmutableMap.of() : properties;
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (closeableGroup != null) {
+      closeableGroup.close();
+    }
+  }
+
+  @Override
+  protected boolean isValidIdentifier(TableIdentifier tableIdentifier) {
+    return tableIdentifier.namespace().levels().length == 1;
+  }
+
+  private String databaseLocation(String dbId) {
+    String warehouseLocation =
+        LocationUtil.stripTrailingSlash(properties.get(CatalogProperties.WAREHOUSE_LOCATION));
+    Preconditions.checkArgument(warehouseLocation != null, "Data warehouse location is not set");
```

I think this would be better if it's checked in `initialize(..)` so that we fail early if this is missing

---

Done.